src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -1073,8 +1073,9 @@ class JoinOnConditionSegment(BaseSegment):
         "ON",
         Indent,
         OneOf(
-            Ref("ExpressionSegment"),
+            # This is a great case for "optionally bracketed"
```

Comment:
See https://github.com/sqlfluff/sqlfluff/blob/master/src/sqlfluff/core/parser/grammar/anyof.py#L226, @barrywhart added it this week!

Reply:
Oh snap! 🎉